contracts/strategies/aura/base/AuraStrategyAuraBAL.sol

```diff
@@ -423,51 +466,85 @@ contract AuraStrategyAuraBAL is
     );
   }
 
-  // Deposit WETH to Balancer and Receive 80BAL20WETH BPT
-  // Deposit 80BAL20WETH to Aura: auraBALBpt Depositor and Receive auraBALBpt
+  /**
+   * @dev There are two ways to receive auraBALBpt
+   *      1. Deposit WETH to Balancer and Receive 80BAL20WETH BPT
+   *         Deposit 80BAL20WETH to Aura: auraBALBpt Depositor and Receive auraBALBpt
+   *      2. Swap WETH to auraBALBpt directly
+   */
   function _convertCrvToAuraBAL() 
     internal 
   {
     address universalRewardToken = rewardToken();
-    uint256 depositTokensAmount = depositNTokens();
     uint256 tokenBalance = IERC20(universalRewardToken).balanceOf(address(this));
 
     IERC20(universalRewardToken).safeApprove(bVault, 0);
     IERC20(universalRewardToken).safeApprove(bVault, tokenBalance);
 
-    // we can accept 0 as minimum, this will be called only by trusted roles
+    address payable recipient = address(uint160(address(this)));
+    address underlyingToken = underlying();
+    bytes32 swapPoolId = balancerSwapPoolId();
+
+    // Try to swap more than deposit
+    uint256 underlyingBefore = IERC20(underlyingToken).balanceOf(address(this));
```

Comment:
@CryptJS13 Hi Jasper, this is how I implement the swapping option instead of direct deposit. My idea is that if I use queryBatchSwap, there might be a risk of having a sandwich attack. So I swap it with a limit. And if the swap succeeds, we will receive more underlying, otherwise, the swap doesn't proceed, and we receive nothing. Then we go with the deposit way.

Reply:
I see, I have adjust it with querySwap before deposit. I was originally afraid if someone finds this transaction in the mempool, they can use the MEV bot to manipulate the price and thus affect the amount we get from the swap.